internal/controller/issues.go

```diff
@@ -0,0 +1,171 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/go-github/v56/github"
+	issues "github.com/matanamar10/github-issue-operator-hhome-assignment/api/v1alpha1"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+	"time"
+)
+
+// searchForIssue checks if GithubIssue CRD has an issue in the repo
+func searchForIssue(issue *issues.GithubIssue, gitHubIssues []*github.Issue) *github.Issue {
+	for _, ghIssue := range gitHubIssues {
+		if ghIssue != nil && strings.EqualFold(*ghIssue.Title, issue.Spec.Title) {
+			return ghIssue
+		}
+	}
+	return nil
+}
+
+// UpdateIssueStatus updates the status of the GithubIssue CRD
+func (r *GithubIssueReconciler) UpdateIssueStatus(ctx context.Context, issue *issues.GithubIssue, githubIssue *github.Issue) error {
+	PRChange := r.CheckForPR(githubIssue, issue)
+	OpenChange := r.CheckIfOpen(githubIssue, issue)
+
+	if PRChange || OpenChange {
+		r.Log.Info("Updating Issue status", zap.String("IssueName", issue.Name), zap.String("Namespace", issue.Namespace))
+
+		if err := r.Client.Status().Update(ctx, issue); err != nil {
+			r.Log.Warn("Status update failed, attempting fallback", zap.Error(err))
+
+			if fallbackErr := r.Client.Update(ctx, issue); fallbackErr != nil {
+				r.Recorder.Event(issue, corev1.EventTypeWarning, "StatusUpdateFailed", fmt.Sprintf("Failed to update status: %v", fallbackErr))
+				return fmt.Errorf("failed to update status: %v", fallbackErr)
+			}
+		}
+
+		r.Recorder.Event(issue, corev1.EventTypeNormal, "StatusUpdated", "Issue status updated successfully")
+		r.Log.Info("Issue status updated successfully", zap.String("IssueName", issue.Name), zap.String("Namespace", issue.Namespace))
+	} else {
+		r.Log.Info("No changes detected in issue status", zap.String("IssueName", issue.Name), zap.String("Namespace", issue.Namespace))
+	}
+
+	return nil
+}
+
+// CheckIfOpen checks if the issue is open
+func (r *GithubIssueReconciler) CheckIfOpen(githubIssue *github.Issue, issueObject *issues.GithubIssue) bool {
+	if githubIssue == nil {
+		return false
+	}
+	state := githubIssue.GetState()
+	condition := &metav1.Condition{
+		Type:    "IssueIsOpen",
+		Status:  metav1.ConditionTrue,
+		Reason:  "IssueIsOpen",
+		Message: "Issue is open",
+	}
+	if state != "open" {
+		condition = &metav1.Condition{
+			Type:    "IssueIsOpen",
+			Status:  metav1.ConditionFalse,
+			Reason:  fmt.Sprintf("IssueIs%s", state),
+			Message: fmt.Sprintf("Issue is %s", state),
+		}
+	}
+	if !meta.IsStatusConditionPresentAndEqual(issueObject.Status.Conditions, "IssueIsOpen", condition.Status) {
+		meta.SetStatusCondition(&issueObject.Status.Conditions, *condition)
+		return true
+	}
+	return false
+}
+
+// CheckForPR checks if the issue has an open PR
+func (r *GithubIssueReconciler) CheckForPR(githubIssue *github.Issue, issueObject *issues.GithubIssue) bool {
```

Comment:
This function is relevant only for `github` so it should be in the `git` package

Reply:
now its generic